### ISSUE TITLE
Can't use Pool and Client together

### ISSUE THREAD

**Issue by planetarydev** [brianc/node-postgres#1483]
Hello,

after some test it seems that I can't use Pool and Client together. By using a single client connection beside a Pool, the pool will return an error with "too much connection" and does'nt wait to callback til a client will released to pool.

The version of the pg-module is 7.3.0

```js
const PGPool = require('pg').Pool;
const PGClient = require('pg').Client;
const async	= require('async');

var client = new PGClient({
	user: DB_USER,
	host: DB_HOST,
	database: DB_DATABASE,
	password: DB_PASSWORD,
	port: DB_PORT
});

var pool = new PGPool({
	user: DB_USER,
	host: DB_HOST,
	database: DB_DATABASE,
	password: DB_PASSWORD,
	port: DB_PORT,
	max: 90
});
var connCount = 0;

// use just a simple client connection outside of the pool
client.connect((err) => {
	if (err) {
		console.error('connection error', err.stack)
	} else {
		console.log('Single Client connected.');
	}
});

// Testing the pool
async.times(100, function(n, next){
	pool.connect((error, connection, release) => {
		// release the first client to pool after 1 second,
		// the second client after 2 seconds...
		setTimeout(()=>{
			release();
		}, 1000 * connCount);

		console.log('get client from pool:', connCount);
		if (error) return next(error);
		connCount++;
		next();
	});
}, function(error, result){
	console.log('Finished with:', connCount);
	console.log(error)
	pool.end();
	done();
});

```

Here is the Error that I received

```
error: tut mir leid, schon zu viele Verbindungen
      at Connection.parseE (node_modules/pg/lib/connection.js:546:11)
      at Connection.parseMessage (node_modules/pg/lib/connection.js:371:19)
      at Socket.<anonymous> (node_modules/pg/lib/connection.js:114:22)
      at addChunk (_stream_readable.js:266:12)
      at readableAddChunk (_stream_readable.js:253:11)
      at Socket.Readable.push (_stream_readable.js:211:10)
      at TCP.onread (net.js:585:20)
```

Without using PGClient the pool will work as expected.

**Comment by charmander** [comment]
> release the first client to pool after 1 second,
> the second client after 2 seconds...

This means acquiring the maximum number of clients from the pool (unless making all the connections takes longer than 9 seconds, which it shouldn’t). The error about the number of connections is from PostgreSQL, not pg. Does it still occur when setting `max: 89`? How about without the client and `max: 91`?

**Comment by planetarydev** [comment]
Thank's a lot! I've found **my** bug and now it's clear

The ConnectionLimit for the database is set to -1 and the max_connections for the server is **100** setup in the postgresql.conf. I'm using pg-admin wich runs in parallel :-) By checking the open connections there are 3 conections displayed in the Dashboard - But this was the Dashboard of the Database - and it shows only the connections to the current db selected and not for the whole server. Changing the view to the server I saw there are 10 connections.

So setting max: 88 will work as expected >> 1x PGClient + 89x PGPool + 10x as show in pg-admin = **100**
